File: proj.android_studio/ee-x-core/src/main/java/com/ee/core/IPlugin.java
package com.ee.core;

import android.app.Activity;
import android.content.Intent;

import androidx.annotation.NonNull;

/**
 * Created by Zinge on 6/1/16.
 */
public interface IPlugin {
    /**
     * Gets the name of the plugin.
     */
    @NonNull
    String getPluginName();

    /**
     * Called when the main activity has been created.
     */
    void onCreate(@NonNull Activity activity);

    /**
     * Called when the main activity has been started.
     */
    void onStart();

    /**
     * Called when the main activity has been stopped.
     */
    void onStop();

    /**
     * Called when the main activity has been resumed.
     */
    void onResume();

    /**
     * Called when the main activity has been paused.
     */
    void onPause();

    /**
     * Called when the main activity has been destroyed.
     */
    void onDestroy();

    /**
     * Called when the application has been destroyed.
     */
    void destroy();

    boolean onActivityResult(int requestCode, int responseCode, Intent data);
}

File: proj.android_studio/ee-x-core/src/main/java/com/ee/core/IPlugin.kt
package com.ee.core

import android.app.Activity
import android.content.Intent
import androidx.annotation.AnyThread
import androidx.annotation.UiThread

/**
 * Created by Zinge on 6/1/16.
 */
interface IPlugin {
    /**
     * Called when the main activity has been created.
     */
    @UiThread
    fun onCreate(activity: Activity)

    /**
     * Called when the main activity has been started.
     */
    @UiThread
    fun onStart()

    /**
     * Called when the main activity has been stopped.
     */
    @UiThread
    fun onStop()

    /**
     * Called when the main activity has been resumed.
     */
    @UiThread
    fun onResume()

    /**
     * Called when the main activity has been paused.
     */
    @UiThread
    fun onPause()

    /**
     * Called when the main activity has been destroyed.
     */
    @UiThread
    fun onDestroy()

    @UiThread
    fun onActivityResult(requestCode: Int, responseCode: Int, data: Intent?): Boolean

    /**
     * Called when the application has been destroyed.
     */
    @AnyThread
    fun destroy()
}